init/findPackageJson.ts

```diff
@@ -0,0 +1,85 @@
+import path from 'path'
+import fs from 'fs'
+import findUp from 'find-up'
+
+type PackageJsonLike = {
+  name?: string
+  scripts?: Record<string, string>
+  dependencies?: Record<string, string>
+  devDependencies?: Record<string, string>
+  [key: string]: unknown
+}
+
+type FindPackageJsonResult =
+  | {
+      packageData: PackageJsonLike
+      filename: string
+      done: false
```

Comment:
also, if it succeeds it has `done: false`?

Reply:
I tried to implement something simmilar to native iterators https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols 

Yes, the `done: true` here means that the iterator is finished the job and in this particular case it means that we didn't find the package.json that we are looking for and came to the root of file system